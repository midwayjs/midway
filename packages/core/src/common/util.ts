import { dirname, resolve, sep, extname } from 'path';

export const safeRequire = p => {
  if (p.startsWith(`.${sep}`) || p.startsWith(`..${sep}`)) {
    p = resolve(dirname(module.parent.filename), p);
  }
  try {
    return require(p);
  } catch (err) {
    return undefined;
  }
};

export const isPath = (p): boolean => {
  // eslint-disable-next-line no-useless-escape
  if (/(^[\.\/])|:|\\/.test(p)) {
    return true;
  }
  return false;
};

/**
 *  safelyGet(['a','b'],{a: {b: 2}})  // => 2
 *  safelyGet(['a','b'],{c: {b: 2}})  // => undefined
 *  safelyGet(['a','1'],{a: {"1": 2}})  // => 2
 *  safelyGet(['a','1'],{a: {b: 2}})  // => undefined
 *  safelyGet('a.b',{a: {b: 2}})  // => 2
 *  safelyGet('a.b',{c: {b: 2}})  // => undefined
 */
export function safelyGet(list: string | string[], obj?: object): any {
  if (arguments.length === 1) {
    return (_obj: object) => safelyGet(list, _obj);
  }

  if (typeof obj === 'undefined' || typeof obj !== 'object' || obj === null) {
    return void 0;
  }
  const pathArrValue = typeof list === 'string' ? list.split('.') : list;
  let willReturn: any = obj;

  for (const key of pathArrValue) {
    if (typeof willReturn === 'undefined' || willReturn === null) {
      return void 0;
    } else if (typeof willReturn !== 'object') {
      return void 0;
    }
    willReturn = willReturn[key];
  }

  return willReturn;
}

/**
 * 剔除 @ 符号
 * @param provideId provideId
 */
export function parsePrefix(provideId: string) {
  if (provideId.includes('@')) {
    return provideId.substr(1);
  }
  return provideId;
}

export function isPathEqual(one: string, two: string) {
  if (!one || !two) {
    return false;
  }
  const ext = extname(one);
  return one.replace(ext, '') === two;
}
