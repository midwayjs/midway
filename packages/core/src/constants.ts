/**
 * 静态参数
 *
 */
export const KEYS = {
  OBJECTS_ELEMENT: 'objects',
  OBJECT_ELEMENT: 'object',
  IMPORT_ELEMENT: 'import',
  PROPERTY_ELEMENT: 'property',
  LIST_ELEMENT: 'list',
  MAP_ELEMENT: 'map',
  ENTRY_ELEMENT: 'entry',
  VALUE_ELEMENT: 'value',
  PROPS_ELEMENT: 'props',
  PROP_ELEMENT: 'prop',
  SET_ELEMENT: 'set',
  CONSTRUCTOR_ARG_ELEMENT: 'constructor-arg',
  REF_ELEMENT: 'ref',
  JSON_ELEMENT: 'json',
  CONFIGURATION_ELEMENT: 'configuration',

  ID_ATTRIBUTE: 'id',
  PATH_ATTRIBUTE: 'path',
  DIRECT_ATTRIBUTE: 'direct',
  AUTOWIRE_ATTRIBUTE: 'autowire',
  ASYNC_ATTRIBUTE: 'async',
  NAME_ATTRIBUTE: 'name',
  REF_ATTRIBUTE: 'ref',
  KEY_ATTRIBUTE: 'key',
  VALUE_ATTRIBUTE: 'value',
  TYPE_ATTRIBUTE: 'type',
  EXTERNAL_ATTRIBUTE: 'external',
  OBJECT_ATTRIBUTE: 'object',
  RESOURCE_ATTRIBUTE: 'resource',
  SCOPE_ATTRIBUTE: 'scope',

  ASPECT_ELEMENT: 'aspect',
  AROUND_ELEMENT: 'around',
  EXPRESSION_ATTRIBUTE: 'expression',
  EXECUTE_ATTRIBUTE: 'execute',
};

export const FUNCTION_INJECT_KEY = 'midway:function_inject_key';
export const MIDWAY_LOGGER_WRITEABLE_DIR = 'MIDWAY_LOGGER_WRITEABLE_DIR';
export const REQUEST_CTX_KEY = 'ctx';
export const REQUEST_OBJ_CTX_KEY = '_req_ctx';
export const HTTP_SERVER_KEY = '_midway_http_server';
export const REQUEST_CTX_LOGGER_CACHE_KEY = '_midway_ctx_logger_cache';

export const ASYNC_CONTEXT_KEY = Symbol('ASYNC_CONTEXT_KEY');
export const ASYNC_CONTEXT_MANAGER_KEY = 'MIDWAY_ASYNC_CONTEXT_MANAGER_KEY';

export const DEFAULT_PATTERN = [
  '**/**.ts',
  '**/**.js',
  '**/**.mts',
  '**/**.mjs',
  '**/**.cts',
  '**/**.cjs',
];

export const IGNORE_PATTERN = ['**/**.d.ts', '**/**.d.mts', '**/**.d.cts'];
