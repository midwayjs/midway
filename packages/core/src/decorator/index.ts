// common
export * from './common/provide';
export * from './common/inject';
export * from './common/pipeline';
export * from './common/aspect';
export * from './common/autoload';
export * from './common/configuration';
export * from './common/objectDef';
export * from './common/framework';
export * from './common/filter';
export * from './common/middleware';
export * from './common/guard';
export * from './common/pipe';
export * from './common/mock';

// faas
export * from './faas/serverlessTrigger';

// web
export * from './web/controller';
export * from './web/paramMapping';
export * from './web/requestMapping';
export * from './web/response';

// other
export * from './constant';
export * from './decoratorManager';

// microservice
export * from './microservice/consumer';
export * from './microservice/provider';
export * from './microservice/rabbitmqListener';
export * from './microservice/kafkaListener';

// rpc
export * from './rpc/hsf';

// task
export * from './task/queue';
export * from './task/task';
export * from './task/taskLocal';
export * from './task/schedule';

// ws
export * from './ws/webSocketController';
export * from './ws/webSocketEvent';
