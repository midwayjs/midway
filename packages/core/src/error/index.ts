export * from './base';
export { HttpStatus, httpError } from './http';
export * from './framework';
