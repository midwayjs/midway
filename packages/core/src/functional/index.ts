export { defineConfiguration } from './configuration';
export * from './hooks';
export * from './interface';
