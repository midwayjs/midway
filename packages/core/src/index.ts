export {
  ObjectIdentifier,
  ObjectDefinitionOptions,
  IManagedInstance,
  ScopeEnum,
  saveClassMetadata,
  attachClassMetadata,
  getClassMetadata,
  saveMethodDataToClass,
  attachMethodDataToClass,
  getMethodDataFromClass,
  listMethodDataFromClass,
  saveMethodMetadata,
  attachMethodMetadata,
  getMethodMetadata,
  savePropertyDataToClass,
  attachPropertyDataToClass,
  getPropertyDataFromClass,
  listPropertyDataFromClass,
  savePropertyMetadata,
  attachPropertyMetadata,
  getPropertyMetadata,
  savePreloadModule,
  listPreloadModule,
  saveModule,
  listModule,
  resetModule,
  clearAllModule,
  getParamNames,
  getProviderId,
  getObjectDefinition,
  classNamed,
  generateProvideId
} from '@midwayjs/decorator';
export * from './interface';
export { ContainerLoader } from './loader';
export { MidwayContainer } from './context/midwayContainer';
export { MidwayRequestContainer } from './context/requestContainer';
export { BaseFramework } from './baseFramework';
export * from './context/providerWrapper';
export * from './common/constants';
export {
  safelyGet,
  safeRequire,
  parsePrefix,
} from './common/util';
export * from './features';
export * from './util/webRouterParam';
export { plainToClass, classToPlain } from 'class-transformer';
