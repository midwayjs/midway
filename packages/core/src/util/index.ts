import { dirname, resolve, sep, posix } from 'path';
import { readFileSync } from 'fs';
import { debuglog } from 'util';

const debug = debuglog('midway:container:util');

export const isDevelopmentEnvironment = env => {
  return ['local', 'test', 'unittest'].includes(env);
};

export const getCurrentEnvironment = () => {
  return process.env['MIDWAY_SERVER_ENV'] || process.env['NODE_ENV'] || 'prod';
};

export const safeRequire = (p, enabledCache = true) => {
  if (p.startsWith(`.${sep}`) || p.startsWith(`..${sep}`)) {
    p = resolve(dirname(module.parent.filename), p);
  }
  try {
    if (enabledCache) {
      return require(p);
    } else {
      const content = readFileSync(p, {
        encoding: 'utf-8',
      });
      return JSON.parse(content);
    }
  } catch (err) {
    debug('SafeRequire Warning', err.message);
    return undefined;
  }
};

/**
 *  safelyGet(['a','b'],{a: {b: 2}})  // => 2
 *  safelyGet(['a','b'],{c: {b: 2}})  // => undefined
 *  safelyGet(['a','1'],{a: {"1": 2}})  // => 2
 *  safelyGet(['a','1'],{a: {b: 2}})  // => undefined
 *  safelyGet('a.b',{a: {b: 2}})  // => 2
 *  safelyGet('a.b',{c: {b: 2}})  // => undefined
 */
export function safelyGet(
  list: string | string[],
  obj?: Record<string, unknown>
): any {
  if (arguments.length === 1) {
    return (_obj: Record<string, unknown>) => safelyGet(list, _obj);
  }

  if (typeof obj === 'undefined' || typeof obj !== 'object' || obj === null) {
    return void 0;
  }
  const pathArrValue = typeof list === 'string' ? list.split('.') : list;
  let willReturn: any = obj;

  for (const key of pathArrValue) {
    if (typeof willReturn === 'undefined' || willReturn === null) {
      return void 0;
    } else if (typeof willReturn !== 'object') {
      return void 0;
    }
    willReturn = willReturn[key];
  }

  return willReturn;
}

/**
 * 剔除 @ 符号
 * @param provideId provideId
 */
export function parsePrefix(provideId: string) {
  if (provideId.includes('@')) {
    return provideId.substr(1);
  }
  return provideId;
}

export function getUserHome() {
  return process.env[process.platform === 'win32' ? 'USERPROFILE' : 'HOME'];
}

export function joinURLPath(...strArray) {
  if (strArray.length === 0) {
    return '';
  }
  let p = posix.join(...strArray);
  p = p.replace(/\/+$/, '');
  if (!/^\//.test(p)) {
    p = '/' + p;
  }
  return p;
}

/**
 * 代理目标所有的原型方法，不包括构造器和内部隐藏方法
 * @param derivedCtor
 * @param constructors
 */
export function delegateTargetPrototypeMethod(
  derivedCtor: any,
  constructors: any[]
) {
  constructors.forEach(baseCtor => {
    Object.getOwnPropertyNames(baseCtor.prototype).forEach(name => {
      if (name !== 'constructor' && !/^_/.test(name)) {
        derivedCtor.prototype[name] = function (...args) {
          return this.instance[name](...args);
        };
      }
    });
  });
}

/**
 * 代理目标原型上的特定方法
 * @param derivedCtor
 * @param methods
 */
export function delegateTargetMethod(derivedCtor: any, methods: string[]) {
  methods.forEach(name => {
    derivedCtor.prototype[name] = function (...args) {
      return this.instance[name](...args);
    };
  });
}

/**
 * 代理目标原型属性
 * @param derivedCtor
 * @param properties
 */
export function delegateTargetProperties(
  derivedCtor: any,
  properties: string[]
) {
  properties.forEach(name => {
    Object.defineProperty(derivedCtor.prototype, name, {
      get() {
        return this.instance[name];
      },
    });
  });
}
