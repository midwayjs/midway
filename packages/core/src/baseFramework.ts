import {
  IConfigurationOptions,
  IMidwayApplication,
  IMidwayBootstrapOptions,
  IMidwayContainer,
  IMidwayFramework,
  MidwayFrameworkType,
  MidwayProcessTypeEnum,
} from './interface';
import { ContainerLoader } from './';
import { APPLICATION_KEY } from '@midwayjs/decorator';

export abstract class BaseFramework<
  APP extends IMidwayApplication,
  T extends IConfigurationOptions
> implements IMidwayFramework<APP, T> {
  protected isTsMode = true;
  protected baseDir: string;
  protected appDir: string;
  protected containerLoader: ContainerLoader;
  public configurationOptions: T;
  public app: APP;

  public configure(options: T): BaseFramework<APP, T> {
    this.configurationOptions = options;
    return this;
  }

  public async initialize(
    options: Partial<IMidwayBootstrapOptions>
  ): Promise<void> {
    this.baseDir = options.baseDir;
    this.appDir = options.appDir;

    this.containerLoader = new ContainerLoader({
      baseDir: this.baseDir,
      isTsMode: this.isTsMode,
      preloadModules: options.preloadModules || [],
    });

    /**
     * initialize containerLoader and initialize ioc container instance
     */
    await this.beforeInitialize(options);
    this.containerLoader.initialize();

    /**
     * load directory and bind files to ioc container
     */
    await this.beforeDirectoryLoad(options);
    const applicationContext = this.containerLoader.getApplicationContext();
    applicationContext.registerObject('baseDir', this.baseDir);
    applicationContext.registerObject('appDir', this.appDir);
    // 如果没有关闭autoLoad 则进行load
    this.containerLoader.loadDirectory(options);

    // register app
    this.containerLoader.registerHook(APPLICATION_KEY, () => {
      return this.getApplication();
    });

    await this.afterDirectoryLoad(options);

    if (!this.app.getApplicationContext) {
      this.app = this.defineApplicationProperties(this.app);
    }

    /**
     * start to load configuration and lifeCycle
     */
    await this.containerLoader.refresh();
    await this.afterInitialize(options);
  }

  public getApplicationContext(): IMidwayContainer {
    return this.containerLoader.getApplicationContext();
  }

  public getConfiguration(key?: string): any {
    return this.getApplicationContext()
      .getConfigService()
      .getConfiguration(key);
  }

  public getCurrentEnvironment() {
    return this.getApplicationContext()
      .getEnvironmentService()
      .getCurrentEnvironment();
  }

  public abstract getFrameworkType(): MidwayFrameworkType;

  public abstract getApplication(): APP;

  public abstract run(): Promise<void>;

  public async stop(): Promise<void> {
    await this.beforeStop();
    await this.containerLoader.stop();
  }

  protected defineApplicationProperties(app: APP): APP {
    return Object.assign(app, {
      getBaseDir: () => {
        return this.baseDir;
      },

      getAppDir: () => {
        return this.appDir;
      },

      getEnv: () => {
        return this.getApplicationContext()
          .getEnvironmentService()
          .getCurrentEnvironment();
      },

      getConfig: (key?: string) => {
        return this.getApplicationContext()
          .getConfigService()
          .getConfiguration(key);
      },

      getFrameworkType: () => {
        return this.getFrameworkType();
      },

      getProcessType: () => {
        return MidwayProcessTypeEnum.APPLICATION;
      },
    });
  }

  protected async beforeStop(): Promise<void> {}

  protected async beforeInitialize(
    options: Partial<IMidwayBootstrapOptions>
  ): Promise<void> {}

  protected async beforeDirectoryLoad(
    options: Partial<IMidwayBootstrapOptions>
  ): Promise<void> {}

  protected async afterDirectoryLoad(
    options: Partial<IMidwayBootstrapOptions>
  ): Promise<void> {}

  protected abstract async afterInitialize(
    options: Partial<IMidwayBootstrapOptions>
  ): Promise<void>;
}
