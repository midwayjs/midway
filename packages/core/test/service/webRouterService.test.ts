import { createLightFramework } from '../util';
import * as path from 'path';
import { MidwayContainer, MidwayWebRouterService, bindContainer, clearAllModule } from '../../src';

describe('/test/service/webRouterService.test.ts', function () {

  it('should test add router first', async () => {
    const collector = new MidwayWebRouterService();
    collector.addRouter(async (ctx) => {
      return 'hello world';
    }, {
      url: '/abc/dddd/efg',
      requestMethod: 'GET',
    });
    const result = await collector.getFlattenRouterTable();
    expect(result).toMatchSnapshot();
  });

  it('should test add router match', async () => {
    const collector = new MidwayWebRouterService();
    collector.addRouter(async (ctx) => {
      return 'hello world';
    }, {
      url: '/',
      requestMethod: 'GET',
    });
    collector.addRouter(async (ctx) => {
      return 'test all method';
    } ,{
      url: '/test/all/method',
      requestMethod: 'ALL',
    });
    collector.addRouter(async (ctx) => {
      return 'hello world';
    }, {
      url: '/abc/dddd/*',
      requestMethod: 'GET',
    });
   
    let routeInfo = await collector.getMatchedRouterInfo('/api', 'get');
    expect(routeInfo).toBeUndefined();

    routeInfo = await collector.getMatchedRouterInfo('/abc/dddd/efg', 'GET');
    expect(routeInfo?.url).toEqual('/abc/dddd/*');

    routeInfo = await collector.getMatchedRouterInfo('/test/all/method', 'POST');
    expect(routeInfo?.url).toEqual('/test/all/method');
    expect(routeInfo?.requestMethod.toUpperCase()).toEqual('ALL');

    collector.addRouter(async (ctx) => {
      return 'hello world';
    }, {
      url: '/*',
      requestMethod: 'GET',
    });
    await collector.getFlattenRouterTable({
      compileUrlPattern: true,
    });
    routeInfo = await collector.getMatchedRouterInfo('/api', 'get');
    expect(routeInfo.url).toEqual('/*')
  });

  it('fix issue 2319', async () => {
    const framework = await createLightFramework(path.join(
      __dirname,
      '../fixtures/issue-2319/src'
    ));
    const collector = await framework.getApplicationContext().getAsync('midwayWebRouterService') as any;
    const fullUrls = (await collector.getFlattenRouterTable()).map(item => {
      return item.fullUrl;
    });
    expect(fullUrls).toMatchSnapshot();
  });

  it('should test generate router and flatten router', async () => {
    const framework = await createLightFramework(path.join(
      __dirname,
      '../fixtures/base-app-controller/src'
    ));
    const midwayWebRouterService = await framework.getApplicationContext().getAsync(MidwayWebRouterService);
    const result = await midwayWebRouterService.getRouterTable();
    expect(result.size).toEqual(3);

    const list = await midwayWebRouterService.getRoutePriorityList();
    expect(list.length).toEqual(3);

    const routes = await midwayWebRouterService.getFlattenRouterTable();
    expect(routes.length).toEqual(14);

    midwayWebRouterService.addRouter(async (ctx) => {
      return 'hello world';
    }, {
      url: '/abc/dddd/*',
      requestMethod: 'GET',
    });

    const routes1 = await midwayWebRouterService.getFlattenRouterTable({
      compileUrlPattern: true,
    });
    expect(routes1.length).toEqual(15);

    const matchedRouterInfo = await midwayWebRouterService.getMatchedRouterInfo('/abc/dddd/efg', 'GET');
    delete matchedRouterInfo.id;
    expect(matchedRouterInfo).toMatchSnapshot();
  });

  it('should test duplicate controller and empty options will not throw', async () => {
    const framework = await createLightFramework(path.join(
      __dirname,
      '../fixtures/app-with-duplicate-controller-not-throw/src'
    ));
    const collector = await framework.getApplicationContext().getAsync(MidwayWebRouterService);
    expect(await collector.getFlattenRouterTable()).toBeDefined();
  });

  it('should test duplicate controller prefix and options', async () => {
    const framework = await createLightFramework(path.join(
      __dirname,
      '../fixtures/app-with-duplicate-controller-options/src'
    ));
    const collector = await framework.getApplicationContext().getAsync(MidwayWebRouterService);
    await expect(collector.getFlattenRouterTable()).rejects.toThrow('duplicated controller options');
  });

  it('should test add controller and with router filter', async () => {
    const framework = await createLightFramework(path.join(
      __dirname,
      '../fixtures/app-with-controller-filter/src'
    ));
    const collector = await framework.getApplicationContext().getAsync(MidwayWebRouterService);
    const result = await collector.getFlattenRouterTable();
    expect(result.length).toEqual(1);
  });

  it('should sort param', function () {
    const collector = new MidwayWebRouterService();
    const result = collector.sortRouter(require('./router').routerList1);
    expect(result[0].url).toEqual('/json');
    expect(result[1].url).toEqual('/');
    expect(result[2].url).toEqual('/:abc/123');
    expect(result[3].url).toEqual('/:fileName');
    expect(result[4].url).toEqual('/abc/*');
    expect(result[5].url).toEqual('/*');
  });

  it('should sort wildcard', function () {
    const collector = new MidwayWebRouterService();
    const result = collector.sortRouter(require('./router').routerList2);
    expect(result[0].url).toEqual('/update');
    expect(result[1].url).toEqual('/');
    expect(result[2].url).toEqual('/*');
  });

  it('fix issue 1008', function () {
    const collector = new MidwayWebRouterService({});
    const result1 = collector.sortRouter(require('./router').routerList3);
    expect(result1[0].url).toEqual('/:page/page');
    expect(result1[1].url).toEqual('/page/:page');
    expect(result1[2].url).toEqual('/:category/:slug');

    const result2 = collector.sortRouter(require('./router').routerList4);
    expect(result2[0].url).toEqual('/page/:page');
    expect(result2[1].url).toEqual('/:page/page');
    expect(result2[2].url).toEqual('/:category/:slug');
  });

  it('should test global prefix', async () => {
    clearAllModule();
    const container = new MidwayContainer();
    bindContainer(container);
    container.bindClass(MidwayWebRouterService);
    container.bindClass(require('../util/fixtures/home'));
    const collector = await container.getAsync(MidwayWebRouterService, [{ globalPrefix: 'api'}]);
    const list = await collector.getRoutePriorityList();
    expect(list.length).toEqual(3);
    expect(list[0].prefix).toEqual('/api/test');
    expect(list[1].prefix).toEqual('/api');
    expect(list[2].prefix).toEqual('/');
    const result = await collector.getFlattenRouterTable();
    expect(result.length).toEqual(4);
    expect(result[3].prefix).toEqual('/');
  });

  it('should test global prefix with router ignore', async () => {
    clearAllModule();
    const container = new MidwayContainer();
    bindContainer(container);
    container.bindClass(MidwayWebRouterService);
    container.bindClass(require('../util/fixtures/prefix-normal'));
    const collector = await container.getAsync(MidwayWebRouterService, [{ globalPrefix: 'api'}]);
    const list = await collector.getRoutePriorityList();
    expect(list.length).toEqual(2);
  });

  it('fix issue 1382', function () {
    const collector = new MidwayWebRouterService();
    const result1 = collector.sortRouter(require('./router').routerList5);
    expect(result1[0].url).toEqual('/detail/:id.html');
    expect(result1[1].url).toEqual('/:typeid/:area/');
  });

  it('test base param router', function () {
    const collector = new MidwayWebRouterService();
    const result1 = collector.sortRouter(require('./router').routerList6);
    expect(result1[0].url).toEqual('/hello');
    expect(result1[1].url).toEqual('/:slot');
  });

});
