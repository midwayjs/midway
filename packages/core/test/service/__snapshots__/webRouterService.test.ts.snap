// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/test/service/webRouterService.test.ts fix issue 2319 1`] = `
[
  "/test/test/asd",
  "/test/asd",
  "/test/**",
  "/**",
]
`;

exports[`/test/service/webRouterService.test.ts should test add router first 1`] = `
[
  {
    "_category": 2,
    "_level": 3,
    "_paramString": "",
    "_pureRouter": "/abc/dddd/efg",
    "_weight": 3,
    "fullUrl": "/abc/dddd/efg",
    "fullUrlFlattenString": "/abc/dddd/efg",
    "method": [Function],
    "requestMethod": "GET",
    "url": "/abc/dddd/efg",
  },
]
`;

exports[`/test/service/webRouterService.test.ts should test generate router and flatten router 1`] = `
{
  "_category": 0,
  "_level": 3,
  "_paramString": "",
  "_pureRouter": "/abc/dddd/",
  "_weight": 2,
  "fullUrl": "/abc/dddd/*",
  "fullUrlCompiledRegexp": /\\^\\\\/abc\\\\/dddd\\(\\?:\\\\/\\(\\.\\*\\)\\)\\[\\\\/#\\\\\\?\\]\\?\\$/i,
  "fullUrlFlattenString": "/abc/dddd/(.*)",
  "method": [Function],
  "requestMethod": "GET",
  "url": "/abc/dddd/*",
}
`;
